opencensus/proto/stats/metrics/metrics.proto

```diff
@@ -215,12 +215,49 @@ message DistributionValue {
     // The number of values in each bucket of the histogram, as described in
     // bucket_bounds.
     int64 count = 1;
-
-    // TODO: Add support for exemplars.
   }
 
   // If the distribution does not have a histogram, then omit this field.
   // If there is a histogram, then the sum of the values in the Bucket counts
   // must equal the value in the count field of the distribution.
   repeated Bucket buckets = 6;
+
+  // Exemplars are example points that may be used to annotate aggregated
+  // Distribution values. They are metadata that gives information about a
+  // particular value added to a Distribution bucket.
+  message Exemplar {
+    // Value of the exemplar point.
+    double value = 1;
+
+    // The observation (sampling) time of the above value.
+    google.protobuf.Timestamp timestamp = 2;
+
+    // The context of a span to be attached to Exemplar.
+    message SpanContext {
+
+      // The resource name of the span.
+      string span_name = 1;
```

Comment:
I think this should be trace ID, not span_name.

How about:
```
// Unique identifier for a trace showing this exemplar.
// The ID is a 16-byte array.
bytes trace_id = 1;
```

Note that https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#Exemplar says attachments[] can be "https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#Exemplar"

And https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v1#google.devtools.cloudtrace.v1.Trace says trace_id is "128-bit numeric value formatted as a 32-byte hex string"

Whereas https://github.com/census-instrumentation/opencensus-proto/blob/master/opencensus/proto/trace/trace.proto has "bytes trace_id" documented to be a "16-byte array"

Reply:
And if we're linking to a span, then there should also be an 8-byte span_id. Otherwise rename this message from SpanContext to TraceSomething.

Reply:
Make sense to me, and yes we need both the trace id and span id in span context.

Please note that the Exemplar attachment we're going to use for trace is the [span name from cloud trace v2](https://cloud.google.com/trace/docs/reference/v2/rest/v2/projects.traces/batchWrite#Span), instead of the [v1 trace_id](https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v1#google.devtools.cloudtrace.v1.Trace):

>The resource name of the span in the following format:

> projects/[PROJECT_ID]/traces/[TRACE_ID]/spans/[SPAN_ID]

>[TRACE_ID] is a unique identifier for a trace within a project; it is a 32-character hexadecimal encoding of a 16-byte array.

>[SPAN_ID] is a unique identifier for a span within a trace; it is a 16-character hexadecimal encoding of an 8-byte array.